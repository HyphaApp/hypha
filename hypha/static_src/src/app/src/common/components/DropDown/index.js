import * as React from "react";
import PropTypes from 'prop-types';

import HelperComponent from "@common/components/HelperComponent";


const DropDown = props => {
  return <div className="form__group ">
    <label htmlFor={props.name} className="form__question form__question--choice_field select" required="">
      <span>{props.label}</span>
      {props.required ? <span className="form__required"> *</span> : ""}
    </label>
    <HelperComponent {...props.helperProps} />
    
    <div className="form__item">
      <div className="form__select">
        <select
          name={props.name}
          id={props.name}
          onChange={e => props.onChange(props.name, e.currentTarget.value)}
        >
          {(props.choices || []).map(choice => {
            return <option key={choice[0]} value={choice[0]} selected={props.value && props.value == choice[0] }>{choice[1]}</option>
          })}
        </select>
      </div>
    </div>
  </div>
}
DropDown.propTypes = {
  name: PropTypes.string,
  label: PropTypes.string,
  onChange: PropTypes.func,
  required: PropTypes.bool,
  choices: PropTypes.array,
  helperProps: PropTypes.object,
  value: PropTypes.node
}

DropDown.displayName = 'DropDown';
export default DropDown;
