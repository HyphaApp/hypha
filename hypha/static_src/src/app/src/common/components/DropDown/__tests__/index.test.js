import React from "react";
import { mount } from "enzyme";
import sinon from "sinon";
import DropDown from "../index";
import * as enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
enzyme.configure({ adapter: new Adapter() });


describe("Test DropDown component", () => {
  const name = "test name";
  const label = "test label";
  const required = false;
  const choices = [["1", "one"], ["2", "two"]];
  const helperProps = {};
  const value = "1";
  const onChange = jest.fn();

  const subject = mount(<DropDown
    name={name}
    label={label}
    required={required}
    choices={choices}
    value={value}
    helperProps={helperProps}
    onChange={onChange}
  />);

  it("Shoud render without issues", () => {
    expect(subject.length).toBe(1);
  });

  it("Should have searched classes and span element with passed text", () => {
    expect(subject.find('.form__group').length).toBe(1);
    expect(subject.find('select').length).toBe(1);
    expect(subject.find('span').text()).toBe(label);
    expect(subject.children().length).toEqual(1);
    subject.find('select').props().onChange({currentTarget : { value : 1}});
    expect(onChange).toHaveBeenCalled()
  });

  test('render a DropDown component', () => {
    expect(subject).toMatchSnapshot();
  });

});

describe("Test DropDown component", () => {
  const name = "test name";
  const label = "test label";
  const required = true;
  const choices = [["1", "one"], ["2", "two"]];
  const helperProps = {};
  const value = "1";
  const onChange = jest.fn();

  const subject = mount(<DropDown
    name={name}
    label={label}
    required={required}
    choices={choices}
    value={value}
    helperProps={helperProps}
    onChange={onChange}
  />);

  it("Shoud render without issues", () => {
    expect(subject.length).toBe(1);
  });

  it("Should have searched classes and span element with passed text", () => {
    expect(subject.find('.form__required').length).toBe(1);
  });

  test('render a DropDown component', () => {
    expect(subject).toMatchSnapshot();
  });

});
