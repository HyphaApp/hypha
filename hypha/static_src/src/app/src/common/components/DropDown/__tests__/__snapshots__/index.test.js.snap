// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DropDown component render a DropDown component 1`] = `
<DropDown
  choices={
    Array [
      Array [
        "1",
        "one",
      ],
      Array [
        "2",
        "two",
      ],
    ]
  }
  helperProps={Object {}}
  label="test label"
  name="test name"
  onChange={
    [MockFunction] {
      "calls": Array [
        Array [
          "test name",
          1,
        ],
      ],
    }
  }
  required={false}
  value="1"
>
  <div
    className="form__group "
  >
    <label
      className="form__question form__question--choice_field select"
      htmlFor="test name"
      required=""
    >
      <span>
        test label
      </span>
    </label>
    <HelperComponent />
    <div
      className="form__item"
    >
      <div
        className="form__select"
      >
        <select
          id="test name"
          name="test name"
          onChange={[Function]}
          value="1"
        >
          <option
            key="1"
            value="1"
          >
            one
          </option>
          <option
            key="2"
            value="2"
          >
            two
          </option>
        </select>
      </div>
    </div>
  </div>
</DropDown>
`;

exports[`Test DropDown component render a DropDown component 2`] = `
<DropDown
  choices={
    Array [
      Array [
        "1",
        "one",
      ],
      Array [
        "2",
        "two",
      ],
    ]
  }
  helperProps={Object {}}
  label="test label"
  name="test name"
  onChange={[MockFunction]}
  required={true}
  value="1"
>
  <div
    className="form__group "
  >
    <label
      className="form__question form__question--choice_field select"
      htmlFor="test name"
      required=""
    >
      <span>
        test label
      </span>
      <span
        className="form__required"
      >
         *
      </span>
    </label>
    <HelperComponent />
    <div
      className="form__item"
    >
      <div
        className="form__select"
      >
        <select
          id="test name"
          name="test name"
          onChange={[Function]}
          value="1"
        >
          <option
            key="1"
            value="1"
          >
            one
          </option>
          <option
            key="2"
            value="2"
          >
            two
          </option>
        </select>
      </div>
    </div>
  </div>
</DropDown>
`;
